compatibility_spark-3/src/test/scala/za/co/absa/hyperdrive/compatibility/impl/TestDeltaCDCToSnapshotWriter.scala

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.compatibility.impl
+
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.apache.spark.sql.streaming.Trigger
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+import za.co.absa.commons.io.TempDirectory
+import za.co.absa.hyperdrive.compatibility.api.DeltaCDCToSnapshotWriterConfiguration
+import za.co.absa.hyperdrive.shared.utils.{FileUtils, SparkTestBase}
+
+import java.sql.Timestamp
+import scala.reflect.io.Path
+
+class TestDeltaCDCToSnapshotWriter extends FlatSpec with MockitoSugar with Matchers with BeforeAndAfterEach with SparkTestBase {
+  private val baseDir: TempDirectory = TempDirectory("TestDeltaCDCToSnapshotWriter").deleteOnExit()
+  private val destinationPath = s"${baseDir.path.toAbsolutePath.toString}/destination"
+  private val checkpointPath = s"${baseDir.path.toAbsolutePath.toString}/checkpoint"
+
+  import spark.implicits._
+
+  private val memoryStream = MemoryStream[CDCEvent](1, spark.sqlContext)
+
+  override def beforeEach(): Unit = {
+    Path(destinationPath).deleteRecursively()
+    Path(checkpointPath).deleteRecursively()
+    memoryStream.reset()
+  }
+
+  behavior of "DeltaCDCToSnapshotWriter"
+
+
+  it should "merge cdc events and create latest snapshot table" in {
+    writeOneInput("/test-data/first-input.csv")
+    getResult should contain theSameElementsAs loadCDCEvents("/test-data/first-expected.csv")
+
+    writeOneInput("/test-data/second-input.csv")
+    getResult should contain theSameElementsAs loadCDCEvents("/test-data/second-expected.csv")
+  }
+
+  def writeOneInput(testedInputPath: String): Unit = {
+    val writer = createDeltaCDCToSnapshotWriter()
+
+    memoryStream.addData(loadCDCEvents(testedInputPath))
+    writer.write(memoryStream.toDF()).processAllAvailable()
+  }
+
+  private def createDeltaCDCToSnapshotWriter(): DeltaCDCToSnapshotWriter = new DeltaCDCToSnapshotWriter(
+    DeltaCDCToSnapshotWriterConfiguration(
+      destination = destinationPath,
+      trigger = Trigger.Once(),
+      checkpointLocation = checkpointPath,
+      partitionColumns = Seq.empty,
+      keyColumn = "id",
+      operationColumn = "eventType",
+      operationDeleteValues = Seq("DL", "FD"),
+      precombineColumns = Seq("timestamp","eventType"),
+      precombineColumnsCustomOrder = Map("eventType" -> Seq("PT", "FI", "RR", "UB", "UP", "DL", "FD")),
+      extraConfOptions = Map.empty[String, String]
+    )
+  )
+
+  private def loadCDCEvents(path: String): Seq[CDCEvent] =
+    CDCEvent.loadFromFile(getClass.getResource(s"/delta-cdc-to-snapshot$path").getPath)
+
+  private def getResult: Seq[CDCEvent] = {
+    import spark.implicits._
+    spark.read.format("delta").load(destinationPath).as[CDCEvent].collect().toSeq
+  }
+}
+
+case class CDCEvent(id: String, value: String, timestamp: Timestamp, eventType: String)
```

Comment:
Maybe it would make sense to extract this to a separate class, since you are reusing this logic in the scd2 test

Reply:
Yes, I would like to do it when I have it merged and do it for both PRs